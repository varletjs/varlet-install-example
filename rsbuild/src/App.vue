<template>
  <var-tooltip content="Hello">
    <var-button type="primary" >Hello</var-button>
  </var-tooltip>
</template>

<style>
body {
  transition: background-color 0.25s;
  color: var(--color-text);
  background-color: var(--color-body);
}
</style>
