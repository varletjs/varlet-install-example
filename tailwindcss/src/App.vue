<template>
  <div class="text-on-primary bg-primary text-md">hello</div>
  <div class="text-on-primary-container bg-primary-container text-lg">world</div>
  <div class="bg-primary sm:bg-info md:bg-warning lg:bg-danger xl:bg-success">varlet</div>
  <div class="bg-hsl-primary/50 text-hsl-text/70">tailwindcss</div>
</template>
