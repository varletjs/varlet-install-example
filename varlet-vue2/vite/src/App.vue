<script setup>
import dark from '@varlet-vue2/ui/es/themes/dark';
import { StyleProvider } from '@varlet-vue2/ui';
import HelloWorld from './components/HelloWorld.vue';

let currentTheme = null;
const toggleThemeHandle = () => {
  currentTheme = currentTheme ? null : dark;
  StyleProvider(currentTheme);
};
</script>

<template>
  <div id="app">
    <img alt="VarLet logo" src="./assets/logo.png" />
    <HelloWorld msg="Welcome to Your varlet-vue2" @toggleTheme="toggleThemeHandle" />
  </div>
</template>

<style scoped>
#app {
  height: 100vh;
  padding-top: 60px;
  font-family: Avenir, Helvetica, Arial, sans-serif;
  color: var(--color-text);
  text-align: center;
  background-color: var(--color-body);
  transition: background-color 0.25s, color 0.25s;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
}
</style>
