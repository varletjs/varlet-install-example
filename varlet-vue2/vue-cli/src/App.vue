<template>
  <div id="app">
    <img alt="VarLet logo" src="./assets/logo.png">
    <HelloWorld msg="Welcome to Your varlet-vue2" @toggleTheme="toggleThemeHandle"/>
  </div>
</template>

<script>
import HelloWorld from './components/HelloWorld.vue'
import dark from '@varlet-vue2/ui/es/themes/dark'
import { StyleProvider } from '@varlet-vue2/ui'

export default {
  name: 'App',
  components: {
    HelloWorld
  },
  data: () => ({
    currentTheme: null
  }),
  methods: {
    toggleThemeHandle() {
      this.currentTheme = this.currentTheme ? null : dark
      StyleProvider(this.currentTheme)
    }
  }
}
</script>

<style>
#app {
  font-family: Avenir, Helvetica, Arial, sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  text-align: center;
  height: 100vh;
  padding-top: 60px;
  transition: background-color 0.25s, color 0.25s;
  color: var(--color-text);
  background-color: var(--color-body);
}
</style>
