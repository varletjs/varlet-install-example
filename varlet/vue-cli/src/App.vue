<script setup lang="ts">
function handleClick() {
  Snackbar('Hello')
}
</script>

<template>
  <var-tooltip content="Say Hello">
    <var-button @click="handleClick">
      Say Hello
    </var-button>
  </var-tooltip>
</template>

<style>
body {
  transition: background-color 0.25s, color 0.25s;
  color: var(--color-text);
  background-color: var(--color-body);
}
</style>
