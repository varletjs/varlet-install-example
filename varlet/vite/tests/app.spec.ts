import App from '../src/App.vue'
import { mount } from '@vue/test-utils'
import { expect, it } from 'vitest'

it('test App.vue mount', () => {
  const wrapper = mount(App)

  expect(wrapper.html()).toMatchSnapshot()
})
