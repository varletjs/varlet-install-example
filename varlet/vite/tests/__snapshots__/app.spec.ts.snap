// Vitest Snapshot v1

exports[`test App.vue mount 1`] = `
"<div class=\\"var-tooltip\\"><button class=\\"var-button var--box var-button--normal var--inline-flex var-button--default var-elevation--2\\" type=\\"button\\">
    <!--v-if-->
    <div class=\\"var-button__content\\"> Say Hello <i class=\\"var-icon var-icon--set var-icon-xml\\" style=\\"transition: transform 0ms;\\"></i><i class=\\"var-icon var-ext-icon var-ext-icon--set var-ext-icon-a\\" style=\\"transition: transform 0ms;\\"></i></div>
  </button>
  <!--teleport start-->
  <!--teleport end-->
</div>"
`;
